packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockEvolution.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {
+  Bar,
+  CartesianGrid,
+  Cell,
+  ChartTooltip,
+  ComposedChart,
+  Legend,
+  Line,
+  Typography,
+  XAxis,
+  YAxis,
+} from '@common/components';
+import { isAfter, isEqual } from '@common/utils';
+import {
+  Box,
+  useFormatDate,
+  useTheme,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { DraftRequestLine } from '../hooks';
+import { useRequestFields } from '../../../api';
+
+export interface StockEvolutionProps {
+  draftLine: DraftRequestLine | null;
+}
+
+type valueWithDate = {
+  date: Date;
+};
+const dateSorter = (a: valueWithDate, b: valueWithDate) => {
+  if (isEqual(a.date, b.date)) return 0;
+  if (isAfter(a.date, b.date)) return 1;
+  return -1;
+};
+
+export const StockEvolution: React.FC<StockEvolutionProps> = ({
+  draftLine,
+}) => {
+  const t = useTranslation('replenishment');
+  const theme = useTheme();
+  const d = useFormatDate();
+  const { maxMonthsOfStock, minMonthsOfStock } = useRequestFields([
+    'maxMonthsOfStock',
+    'minMonthsOfStock',
+  ]);
+
+  const data =
+    draftLine?.chartData?.stockEvolution.nodes
+      .map(node => ({
```

Comment:
the data shape is not ideal, so I've had to map it

Reply:
Yeah not why the data shape was changed 🤔 

Maybe share this shape with andrei to get it changed 🤷 the query is specifically for this modal only so no sense in having to map it after receiving